Firebase/Core/FIRLogger.m

```diff
@@ -190,7 +197,12 @@ void FIRSetLoggerLevel(FIRLoggerLevel loggerLevel) {
 #ifdef DEBUG
 void FIRResetLogger() {
   sFIRLoggerOnceToken = 0;
-  [[NSUserDefaults standardUserDefaults] removeObjectForKey:kFIRPersistedDebugModeKey];
+  [sFIRLoggerUserDefaults removeObjectForKey:kFIRPersistedDebugModeKey];
+  sFIRLoggerUserDefaults = nil;
+}
+
+void FIRSetLoggerUserDefaults(NSUserDefaults *defaults) {
```

Comment:
I think this won't work if DEBUG isn't on.

Reply:
I was going to only use it for unit testing at the moment, similar to `FIRResetLogger()`. If we were to move to a separate user defaults system for Core or all of Firebase (which we probably should? we can discuss that separately) then we could move it out of the `#ifdef DEBUG` check.

Does that make sense, or is there a different concern?